### PR TITLE
Implement vpcnetworkconfig crd

### PR DESCRIPTION
**Pull request by lxiaopei**

### CODE REVIEW

**Comment by timdengyun** [comment]
please merge patch to vpc_dev instead of main branch

**Comment by seanpang-vmware** [comment]
Should we have a VPCNetworkConfigStore to save all the configs for others to query?

**Comment by zhengxiexie** [comment]
I see that there is cache.go in vpcnetworkconfig, why not use store.go like other CR?

**Comment by zhengxiexie** [comment]
As discussed, we could deprecate this PR, and put all the logic in the VPC controller as the security policy does.[https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/controllers/securitypolicy/securitypolicy_controller.go#L243](url)

File: pkg/controllers/vpcnetworkconfig/vpcnetworkconfiguration_controller.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	var vpcNetworkConfig v1alpha1.VPCNetworkConfiguration
+	err := r.Client.Get(ctx, req.NamespacedName, &vpcNetworkConfig)
+	var isDeleted bool
+	if err != nil {
+		isDeleted = apierrors.IsNotFound(err)
+		if !isDeleted {
+			klog.InfoS("Unable to fetch VPCNetworkConfiguration", "vpcNetworkConfiguration", req.NamespacedName.String(), "err", err)
+			return ctrl.Result{}, err
+		} else {
+			vpcNetworkConfigurationObj, exist := r.installedVPCNetworkConfigs.GetByKey(req.NamespacedName.String())
```
Comment (TaoZou1) [inline pkg/controllers/vpcnetworkconfig/vpcnetworkconfiguration_controller.go:49]: logic here is different with other controller. The other controller will return error if it is not IsNotFound(err). Any reason here?

File: pkg/controllers/vpcnetworkconfig/cache.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package vpcnetworkconfig
```
Comment (TaoZou1) [inline pkg/controllers/vpcnetworkconfig/cache.go:4]: There is no store.go for this controller, how about other controller needs to access the cache?

Comment (timdengyun) [reply pkg/controllers/vpcnetworkconfig/cache.go:4]: please hold off review,  it's possible to move this PR to vpc_dev branch merge.

**Comment by lxiaopei** [comment]
After discussion with @seanpang-vmware  and @zhengxiexie , this PR is not needed anymore. We will watch VPCNetworkConfiguration CRD in VPC controller.

**Pull request closed without merging**